### PR TITLE
Mat-348: Wire in SVD implementation.

### PR DESCRIPTION
**Pull request by stuartarchibald**
This PR is to add in the actual implementation of the SVD node to replace the dummy behaviour that was put in place as a test.
- Performs wiring to `xgesvd()`.
- Adds unit tests including
  - Reconstruction tests
  - Behaviour tests

BB Multi-Pass: [http://devmaths-lx-1:8011/builders/nyqwk2-testing/builds/797](http://devmaths-lx-1:8011/builders/nyqwk2-testing/builds/797)

Coverage:
Java: 85% (up a bit)
build/src/jshim 0.8% (no change)
build/src/librdag 13.5% (down a bit)
src/jshim       62.5% (no change)
src/librdag     91.2% (down a bit)
src/librdag/runners 100.0% (up a lot)
src/librdag/test/nodes 96.4% (down a bit)

### CODE REVIEW

File: src/generator/dispatchtemplates.py [hunk 1]
```diff
@@ -441,7 +442,9 @@ class DispatchOp
   {
 %(eval_cases)s
     default:
-      throw rdag_error("Unknown type in dispatch on arg0");
+        stringstream message;
+        message << "Unknown type in dispatch on arg0. Type is: " << arg0->getType() << ".";
+        throw rdag_error(message.str());
```
Comment (gmarkall) [inline src/generator/dispatchtemplates.py:447]: This is a much better message than before :-)

Comment (stuartarchibald) [inline src/generator/dispatchtemplates.py:447]: I agree. We should probably create a reverse dictionary at some point.

File: src/librdag/runners/norm2runner.cc [hunk 2]
```diff
@@ -33,10 +33,12 @@
 
 template<typename T>
 void
-dense_runner(RegContainer * reg, OGMatrix<T> const * arg)
+norm2_dense_runner(RegContainer * reg, OGMatrix<T> const * arg)
 {
   const OGRealScalar* ret = nullptr; // the returned item
 
+  arg->debug_print();
```
Comment (gmarkall) [inline src/librdag/runners/norm2runner.cc]: Stray debug print?

Comment (stuartarchibald) [inline src/librdag/runners/norm2runner.cc]: Yes. Well spotted. Patched in b4e6502. 

BB Multi-Pass: [http://devmaths-lx-1:8011/builders/nyqwk2-testing/builds/798](http://devmaths-lx-1:8011/builders/nyqwk2-testing/builds/798)

**Comment by gmarkall** [comment]
This looks great - I like that we have reconstruction tests as well, which are a nice validation of the results. 

One thing that I find takes some mental gymnastics to think about is `ConcreteDenseMatrixFactory()` - that sounds like a class that would have some makeXXX() methods, but actually its a function that returns a dense matrix - perhaps `makeConcreteDenseMatrix()` is more appropriate?

**Comment by stuartarchibald** [comment]
It is just a temporary measure until the dispatcher becomes template aware. However, I see what you mean and will sort it, it's just a quick patch.

**Comment by stuartarchibald** [comment]
Patch 98f403a in response to https://github.com/OpenGamma/nyqwk2/pull/80#issuecomment-38831204

BB Multi-Pass: [http://devmaths-lx-1:8011/builders/nyqwk2-testing/builds/800](http://devmaths-lx-1:8011/builders/nyqwk2-testing/builds/800)

**Comment by gmarkall** [comment]
That's great! Merging.

**Pull request merged**